components/camera/client.go

```diff
@@ -60,11 +60,9 @@ func (c *client) Read(ctx context.Context) (image.Image, func(), error) {
 		return nil, nil, err
 	}
 	if actualType != resp.MimeType {
-		c.logger.Debugw("got different MIME type than what was asked for", "sent", actualType, "received", resp.MimeType)
-	} else {
-		resp.MimeType = mimeType
+		c.logger.Debugw("got different MIME type than what was asked for", "sent", mimeType, "received", resp.MimeType)
 	}
-	img, err := rimage.DecodeImage(ctx, resp.Image, resp.MimeType, int(resp.WidthPx), int(resp.HeightPx))
+	img, err := rimage.DecodeImage(ctx, resp.Image, mimeType)
```

Comment:
should Read not be able to return a lazily encoded image?

Reply:
Hmm I'm not sure, since I guess we decode the image here. But my concern is that we are passing in the wrong mime type here (in case GetImage returned a different mime type from the one we requested).